src/index.tsx

```diff
@@ -73,11 +72,38 @@ var createWebmarkFolder = (): void => {
     )
 };
 
-var getCurrentUrl = (): string => {
-    return 'https://current.url.com';
+// type checker for proper string type url
+let typeCheckString = (text: string | undefined): string => {
+    if (text == undefined) {
+        // TODO: this block needs proper type guard rather than returning below url.
+        return "https://example.com/error";
+    } else {
+        return text;
+    }
 };
 
-var saveToWebmarkFolder = (url: string): void => {
+let getCurrentUrlAndSave = () => {
+    chrome.tabs.query(
+        { active: true, currentWindow: true },
+        ([currentTab]) => {
+            const thisTabUrl: string|undefined = currentTab.url;
+            const thisTabTitle: string|undefined = currentTab.title;
+            saveToWebmarkFolder(thisTabUrl, thisTabTitle);
+        });
```

Comment:
Let's put this into a new line for a while and reconsider our rule once it gets too long (vertically). You can sneak this edit into another commit/issue.

Reply:
I think there was a misunderstanding here. I was suggesting two separate lines for `}` and `);`

Reply:
yeah why didn't you mention that explicitly?

Reply:
I think GitHub always shows three additional lines above to show the context. I'll be more specific from now on.